flow/texture_unittests.cc

```diff
@@ -20,26 +20,111 @@ class MockTexture : public Texture {
              bool freeze,
              GrContext* context) override {}
 
-  void OnGrContextCreated() override {}
-
-  void OnGrContextDestroyed() override {}
-
+  void OnGrContextCreated() override { gr_context_created_ = true; }
+  void OnGrContextDestroyed() override { gr_context_destroyed_ = true; }
   void MarkNewFrameAvailable() override {}
-
   void OnTextureUnregistered() override { unregistered_ = true; }
 
+  bool gr_context_created() { return gr_context_created_; }
+  bool gr_context_destroyed() { return gr_context_destroyed_; }
   bool unregistered() { return unregistered_; }
 
  private:
+  bool gr_context_created_ = false;
+  bool gr_context_destroyed_ = false;
   bool unregistered_ = false;
 };
 
-TEST(TextureRegistry, UnregisterTextureCallbackTriggered) {
-  TextureRegistry textureRegistry;
-  std::shared_ptr<MockTexture> mockTexture = std::make_shared<MockTexture>(0);
-  textureRegistry.RegisterTexture(mockTexture);
-  textureRegistry.UnregisterTexture(0);
-  ASSERT_TRUE(mockTexture->unregistered());
+class TextureRegistryTest : public ::testing::Test {
+ public:
+  TextureRegistryTest() = default;
+  ~TextureRegistryTest() override = default;
+
+  TextureRegistry& registry() { return registry_; }
+
+ private:
+  TextureRegistry registry_;
+};
+
+TEST_F(TextureRegistryTest, UnregisterTextureCallbackTriggered) {
+  std::shared_ptr<MockTexture> mock_texture1 = std::make_shared<MockTexture>(0);
```

Comment:
Nit: Here and elsewhere. `auto mock_texture1` (you seemed to prefer this earlier on and I do too).